editor.planx.uk/src/@planx/components/PlanningConstraints/Public.test.tsx

```diff
@@ -55,18 +58,137 @@ describe("error state", () => {
           fn="property.constraints.planning"
           disclaimer="This page does not include information about historic planning conditions that may apply to this property."
         />
-        ,
       </ErrorBoundary>,
     );
     const results = await axe(container);
     expect(results).toHaveNoViolations();
   });
 });
 
-it.todo("renders correctly");
+describe("following a FindProperty component", () => {
+  beforeAll(() => {
+    act(() =>
+      useStore.setState({
+        breadcrumbs: simpleBreadcrumbs,
+        flow: simpleFlow,
+        teamIntegrations: {
+          hasPlanningData: true,
+        },
+      }),
+    );
+  });
+
+  it("renders correctly", async () => {
+    const handleSubmit = vi.fn();
+
+    const { user, getByRole, getByTestId } = setup(
+      <PlanningConstraints
+        title="Planning constraints"
+        description="Things that might affect your project"
+        fn="property.constraints.planning"
+        disclaimer="This page does not include information about historic planning conditions that may apply to this property."
+        handleSubmit={handleSubmit}
+      />,
+    );
+
+    expect(
+      getByRole("heading", { name: "Planning constraints" }),
+    ).toBeInTheDocument();
+
+    await user.click(getByTestId("continue-button"));
+
+    expect(handleSubmit).toHaveBeenCalled();
+  });
+
+  it("should not have any accessibility violations", async () => {
+    const { container } = setup(
+      <PlanningConstraints
+        title="Planning constraints"
+        description="Things that might affect your project"
+        fn="property.constraints.planning"
+        disclaimer="This page does not include information about historic planning conditions that may apply to this property."
+      />,
+    );
+    const results = await axe(container);
+    expect(results).toHaveNoViolations();
+  });
+
+  it("fetches planning constraints when we have lng, lat or siteBoundary", async () => {
+    setup(
+      <PlanningConstraints
+        title="Planning constraints"
+        description="Things that might affect your project"
+        fn="property.constraints.planning"
+        disclaimer="This page does not include information about historic planning conditions that may apply to this property."
+        handleSubmit={vi.fn()}
+      />,
+    );
+
+    expect(swr).toHaveBeenCalled();
+
+    // Planning data is called first
+    const swrURL = swrMock.calls[0][0]();
+    const swrResponse = swrMock.results[0].value;
+
+    expect(swrURL).toContain("/gis");
+    expect(swrResponse).toEqual({ data: digitalLandResponseMock });
+  });
+
+  it("fetches classified roads only when we have a siteBoundary", () => {
```

Comment:
nit: roads fetch relies on USRN, not `siteBoundary` - I think description is a bit misguided/outdated here ?  

I _think_ better test scenarios are:
- If user selected an OS address & has a USRN, we can check/fetch roads
- If user proposed a new address & only has a lat/long, we cannot check/fetch roads (but we can still check planning data)

Reply:
Fixed in [e0dd5f8](https://github.com/theopensystemslab/planx-new/pull/3700/commits/e0dd5f821625e4d15d4b8b12f199925ab069cf22)